src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,307 @@
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(LINE);
+    }
+
+    /**
+     * Prints the welcome message when user first runs the program.
+     */
+    public static void printWelcomeMessage() {
+        System.out.println(LINE);
+        printLogoMessage();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+    }
+
+    /**
+     * Prints the logo of the bot.
+     */
+    public static void printLogoMessage() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(logo);
+    }
+
+    /**
+     * Print a format error message with the given format usage.
+     *
+     * @param format Constant string containing the format usage of the command.
+     */
+    public static void printFormatErrorMessage(String format) {
+        System.out.println("Error: Incorrect format detected.");
+        System.out.println("Usage: " + format);
+    }
+
+    /**
+     * Method to process user command accordingly.
+     *
+     * @param fileHandler FileHandler to contain and extract tasks list information.
+     * @param taskManager TaskManager object that handles all task related action.
+     * @param userInputs  Raw user inputs from scanner.
+     * @return Whether if the exit command has been triggered.
+     */
+    public static boolean processInput(FileHandler fileHandler, TaskManager taskManager, String userInputs) {
+        boolean isNotExit = true;
+        boolean hasChange = false;
+        System.out.println(LINE);
+        switch (getUserCommand(userInputs)) {
+        case ADD_TODO_COMMAND:
+            addToDoTask(taskManager, userInputs);
+            hasChange = true;
+            break;
+        case ADD_EVENT_COMMAND:
+            addEventTask(taskManager, userInputs);
+            hasChange = true;
+            break;
+        case ADD_DEADLINE_COMMAND:
+            addDeadlineTask(taskManager, userInputs);
+            hasChange = true;
+            break;
+        case SET_DONE_COMMAND:
+            try {
+                int taskIndex = getTaskIndexFromUserInputs(userInputs);
+                taskManager.setTaskToDone(taskIndex);
+                hasChange = true;
+            } catch (InvalidTaskNumberException e) {
+                printFormatErrorMessage(DONE_FORMAT);
+            }
+            break;
+        case DELETE_COMMAND:
+            try {
+                int taskIndex = getTaskIndexFromUserInputs(userInputs);
+                taskManager.deleteTask(taskIndex);
+                hasChange = true;
+            } catch (InvalidTaskNumberException e) {
+                printFormatErrorMessage(DELETE_FORMAT);
+            }
+            break;
```

Comment:
Consider applying SLAP on these statements is consistent in their abstraction level for this code fragment.